aquadoggo/src/materializer/tasks/blob.rs

```diff
@@ -75,38 +71,33 @@ pub async fn blob_task(context: Context, input: TaskInput) -> TaskResult<TaskInp
             // We don't raise a critical error here, as it is possible that this method returns an
             // error.
             .map_err(|err| TaskError::Failure(err.to_string()))?
-            .unwrap();
+            .expect("Blob data exists at this point");
 
         // Compose, and when needed create, the path for the blob file.
         let base_path = match &context.config.blob_dir {
             Some(base_path) => base_path,
             None => return Err(TaskError::Critical("No base path configured".to_string())),
         };
-
-        let blob_dir = base_path.join(blob_document.id().as_str());
-
-        fs::create_dir_all(&blob_dir).map_err(|err| TaskError::Critical(err.to_string()))?;
-        let blob_view_path = blob_dir.join(blob_document.view_id().to_string());
+        let blob_view_path = base_path.join(blob_document.view_id().to_string());
 
         // Write the blob to the filesystem.
-        info!("Creating blob at path {blob_view_path:?}");
-
-        let mut file = File::create(&blob_view_path).unwrap();
-        file.write_all(blob_data.as_bytes()).unwrap();
-
-        // create a symlink from `../documents/<document_id>` -> `../<document_id>/<current_view_id>`
-        if is_current_view(&context.store, blob_document.view_id()).await? {
-            info!("Creating symlink from document id to current view");
-
-            let link_path = base_path
-                .join(BLOBS_SYMLINK_DIR_NAME)
-                .join(blob_document.id().as_str());
-
-            let _ = fs::remove_file(&link_path);
-
-            symlink(blob_view_path, link_path)
-                .map_err(|err| TaskError::Critical(err.to_string()))?;
-        }
+        info!("Creating blob at path {}", blob_view_path.display());
+
+        let mut file = File::create(&blob_view_path).await.map_err(|err| {
+            TaskError::Critical(format!(
+                "Could not create blob file @ {}: {}",
+                blob_view_path.display(),
+                err
+            ))
+        })?;
+
+        file.write_all(&blob_data).await.map_err(|err| {
+            TaskError::Critical(format!(
+                "Could not write blob file @ {}: {}",
+                blob_view_path.display(),
+                err
+            ))
+        })?;
```

Comment:
I've changed this to use the `tokio::fs` methods, this assures that writing to files is done in a separate thread and doesn't block the main thread :+1: